src/atom.rs

```diff
@@ -123,7 +123,7 @@ fn write_atom_feed_from_records_impl<W>(
     collection_name: String,
     type_namespace: String,
     type_name: String,
-) -> std::result::Result<(), quick_xml::Error>
+) -> std::result::Result<(), ODataError>
```

Comment:
The primary reason I separated the two `write_*_impl` functions was to avoid adding automatic conversion from `quick_xml::Error` into `ODataError`, but alas...

Let's go with automatic conversion for now, but can you please inline these functions back into their parents:
- `write_atom_feed_from_records_impl` should be inlined back into `write_atom_feed_from_records`
- `write_atom_entry_from_record_impl` should be inlined back into `write_atom_entry_from_record`

Reply:
I think the proper way to separate the errors would be:
- First check the schema for compatibility
  - This is also where we can chose to either error-out on unsupported types or skip unsupported columns (depending on the `OnUnsupported` enum value)
- Only after that do the encoding, freely panicking on unsupported types (as those should've never gotten this far)

What do you think about this alternative approach @hozan23?

Reply:
I'm not sure why you're against the automatic conversion for the errors. Many Rust libraries, including `Datafusion`, use this approach, so I believe it would be good to follow that practice, especially since this repository builds on top of `Datafusion`. Users would likely expect the same approach, as I did, to be honest. It's tedious to manage multiple error types from a single library.

Reply:
> I think the proper way to separate the errors would be:
> 
>     * First check the schema for compatibility
>       
>       * This is also where we can chose to either error-out on unsupported types or skip unsupported columns (depending on the `OnUnsupported` enum value)
> 
>     * Only after that do the encoding, freely panicking on unsupported types (as those should've never gotten this far)

Can you elaborate here? Should we check the schema compatibility with the `to_edm_type` function in `odata_collection_handler` as well?

Reply:
> I'm not sure why you're against the automatic conversion for the errors. Many Rust libraries, including Datafusion, use this approach

Packing all errors into "all-in-one" error like `DataFusionError` may be OK **within** one library ... although after working with DF for two years I can say that it makes it quite hard to map errors into nice user-friendly messages.

In this case we are working on the border of several domains: DataFusion and HTTP/OData and this requires very precise control of errors and status codes. In one contest a missing table should result in `404 Not Found` while in another it has to be treated as `500 Internal Error`.

Automatic conversion makes it way to easy to put `?` everywhere without thinking about the greater context and what error code this will result in API.

Reply:
> Packing all errors into "all-in-one" error like `DataFusionError` may be OK **within** one library ... although after working with DF for two years I can say that it makes it quite hard to map errors into nice user-friendly messages.
> 
> In this case we are working on the border of several domains: DataFusion and HTTP/OData and this requires very precise control of errors and status codes. In one contest a missing table should result in `404 Not Found` while in another it has to be treated as `500 Internal Error`.
> 
> Automatic conversion makes it way to easy to put `?` everywhere without thinking about the greater context and what error code this will result in API.

I think what you are trying to sense makes sense to me now, especially with http errors involved, I know it is hard to map the errors to http errors with the right context, and I don't have too much experience with http libraries but remember having the same problem you mention.